### ISSUE TITLE
File Upload doesn't work properly if uploading files within a Card Container

### ISSUE THREAD

**Issue by mradamcox** [archesproject/arches#2861]
_after trying to tackle this problem, I'm reworking this ticket a little bit with more information... my initial theory involving triggering nodegroups was incorrect._

### User Story
Uploading a file does not work properly in the following scenario.

**A**
![image](https://user-images.githubusercontent.com/10427268/34366322-b0ffab20-ea5f-11e7-9f34-33af02af328d.png)

When I add a photo, I click "Add" which stages the photo info and activates the "Save" button.

**B**
![image](https://user-images.githubusercontent.com/10427268/34366341-e986553e-ea5f-11e7-889b-ad5409720007.png)

When I click Save now, things look fine until I go to the Resource Report, where there is no photo shown.

**C**
![image](https://user-images.githubusercontent.com/10427268/34366358-1104dd38-ea60-11e7-9a79-54f957b2f382.png)

By looking in my app, I can confirm that the file has not been uploaded at all, as if the function was not run.

However, if I go back to the resource editor, open the previously saved image section, remove the photo by clicking the big red X box, and re-add the image, a Save button shows up _within_ the photo section indicating that a change has been made in the tile.

**D**
![image](https://user-images.githubusercontent.com/10427268/34366440-1f67ba3e-ea61-11e7-982e-d2894551328b.png)

When I click save this time, the re-added file is actually uploaded, as you can see in the resource report.

**E**
![image](https://user-images.githubusercontent.com/10427268/34366471-9ba10312-ea61-11e7-86df-f7529345e7b7.png)

By doing some testing, I know now that the function is properly run on the photo upload nodegroup, however, the problem is that when "Save" is clicked on the card container, in the request that gets sent to the function there is no file... in other words, `request.FILES` is an empty list, as returned here: https://github.com/archesproject/arches/blob/master/arches/app/datatypes/datatypes.py?utf8=%E2%9C%93#L836.

This is not the case when the file is added individually to the file-upload widget, as shown in image **D** above. So, current thinking is that while the file widget works fine, the overarching "card container" widget (perhaps `https://github.com/archesproject/arches/blob/master/arches/app/media/js/views/resource/editor/form.js`) does not properly retrieve the list of uploaded tiles and pass it on to the request.

#### Your Arches Information
* Version used: 4.0.1
* Browser Name and version: Chrome/Firefox

#### Context
This issue creates the need for a prohibitively (IMO) cumbersome workaround.

**Comment by mradamcox** [comment]
I'd add that the card container itself doesn't show up as an option in the "triggering nodegroups" definition for the function. If it did, the function could be modified to run every time the card container nodegroup is saved (as well as the specific file nodegroup) which would probably be a decent workaround.

**Comment by mradamcox** [comment]
Turns out that n-n tiles within a card container are not saved when they are added to the card container, only when the card container itself is saved. This works great for everything, however, the file upload data is lost in the process. I made this commit to my own fork https://github.com/legiongis/arches/commit/8d58927d90b91bf5d470331d94bf47dc86cd0d60 to force the save of a tile any time it is added to the card container. This is a temporary fix that will work for FPAN, but it should be improved before going into the main arches repo.
